src/arena.rs

```diff
@@ -118,11 +127,22 @@ impl<T> Handle<T> {
 )]
 #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 pub struct Range<T> {
-    inner: ops::Range<u32>,
+    pub(crate) inner: ops::Range<u32>,
```

Comment:
Since we've already added a `check_contains_handle` method to `Arena`, let's also add a `check_contains_range` method, let `valid::handles`' `Range::check_valid_for` method can just call that, and let this stay private.